modules/engine/src/model/model.ts

```diff
@@ -277,6 +288,8 @@ export class Model {
   }
 
   destroy(): void {
+    this.shaderFactory.release(this.pipeline.vs);
+    this.shaderFactory.release(this.pipeline.fs);
     this.pipelineFactory.release(this.pipeline);
```

Comment:
Somehow releasing the pipeline first seems more safe, since it might still be referencing those destroyed shaders?

Reply:
Ah, makes sense yes!